source/Octopus.Cli/Commands/Runbooks/RunbookCommandBase.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Octopus.Cli.Repositories;
+using Octopus.Cli.Util;
+using Octopus.Client;
+using Octopus.Client.Model;
+using Octopus.CommandLine;
+using Octopus.CommandLine.Commands;
+
+namespace Octopus.Cli.Commands.Runbooks {
+    
+    /// <summary>
+    /// Base class for Runbook related commands with shared logic for all Runbook (and RunbookRun) commands
+    /// </summary>
+    public abstract class RunbookCommandBase : ApiCommand {
+        protected readonly HashSet<string> projects = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        protected IDictionary<string, ProjectResource> projectsById;
+        protected string[] projectsFilter;
+
+        public RunbookCommandBase(IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFileSystem fileSystem, IOctopusClientFactory clientFactory, ICommandOutputProvider commandOutputProvider)
+            : base(clientFactory, repositoryFactory, fileSystem, commandOutputProvider) {
+            var options = Options.For("Listing");
+            options.Add<string>("project=", "[Optional] Name of a project to filter by. Can be specified many times.", v => projects.Add(v), allowsMultiple: true);
```

Comment:
I believe this should be required (not optional). It is possible to have thousands of projects and making this optional runs the risk of very long load-times. If users wish to iterate over all runbooks for all projects then I would prefer to have this scripted as an opt-in.

Reply:
--projectName has been made required as suggested.